interval/query.go

```diff
@@ -0,0 +1,110 @@
+package interval
+
+import (
+	"github.com/vertgenlab/gonomics/axt"
+	"github.com/vertgenlab/gonomics/bed"
+	"github.com/vertgenlab/gonomics/chain"
+	"github.com/vertgenlab/gonomics/sam"
+	"github.com/vertgenlab/gonomics/vcf"
+	"path"
+)
+
+func LiftIntervalWithChain(c *chain.Chain, i Lift) (string, int, int) {
+	return c.QName, chain.TPosToQPos(c, i.GetChromStart()), chain.TPosToQPos(c, i.GetChromEnd())
+}
+
+func GoReadToLiftChan(inputFile string) <-chan Lift {
+	answer := make(chan Lift, 1000)
+	go ReadToLiftChan(inputFile, answer)
+	return answer
+}
+
+func ReadToLiftChan(inputFile string, send chan<- Lift) {
+	// How the file is read is dependent on the file extension
+	filetype := path.Ext(inputFile)
+
+	if filetype == ".gz" {
+		// If terminal extension is ".gz" then trim off the gz and get the next extension
+		filetype = path.Ext(inputFile[0 : len(inputFile)-len(filetype)])
+	}
+
+	switch filetype {
+	case ".bed":
```

Comment:
Bed seems like the "easy" one to implement and I know you are doing vcf for your own work.  These others seem pretty hard, so I don't think we should fully tackle them until someone needs them.  Should we initially leave them commented out, or is it worth it to do a crude lift and with a warning that only coordinates are lifting and not all the underlying data in the struct?

Reply:
Yes, bed should be the simplest case and more widely used